src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2138,10 +2208,11 @@ static List *
 ExtractInsertValuesList(Query *query, Var *partitionColumn)
 {
 	List *insertValuesList = NIL;
-	TargetEntry *targetEntry = get_tle_by_resno(query->targetList,
-												partitionColumn->varattno);
+	RangeTblEntry *valuesRTE = NULL;
 
-	if (targetEntry == NULL)
+	int partitionColumnIndex = GetTargetListEntryIndexByResno(query->targetList,
+															  partitionColumn->varattno);
+	if (partitionColumnIndex == -1)
```

Comment:
Indeed, This particular change is because the partition column's target entry may contain an arbitrary expression which does not refer to values_lists. However, after values_lists expansion we know that the index of the target entry matches the column index in the values_lists.